Content.Server/Storage/Components/EntityStorageComponent.cs

```diff
@@ -183,7 +183,9 @@ public virtual bool CanOpen(EntityUid user, bool silent = false)
         {
             if (IsWeldedShut)
             {
-                if (!silent) Owner.PopupMessage(user, Loc.GetString("entity-storage-component-welded-shut-message"));
+                if (!user.IsInContainer())
+                    if (!silent) Owner.PopupMessage(user, Loc.GetString("entity-storage-component-welded-shut-message"));
```

Comment:
Just combine the checks and also check if user is in the same container?

Reply:
Yea, that'll look better. Thanks!

Reply:
You just combined them, but didn't actually check if the user is in the same container that they are attempting to open (just any container). Maybe just use `Contents.Contains(user)`.

Reply:
Ah yea, goes to show I should probably actually read the function and not assume from it's name what it's going to return.. Thanks